rusha.sweet.js

```diff
@@ -509,16 +509,51 @@
   // messages containing a jobid and a buffer
   // or blob object, and return the hash result.
   if (typeof FileReaderSync !== 'undefined') {
-    var reader = new FileReaderSync(),
-        hasher = new Rusha(4 * 1024 * 1024);
-    self.onmessage = function onMessage (event) {
-      var hash, data = event.data.data;
+    var hasher;
+    var readTotal = 0;
+    var BLOCK_SIZE;
+    var hashData = function hash (data, cb) {
```

Comment:
Could we pass in the `hasher` here and in `hashFile` as an argument instead of making it a variable in the outer scope? That seems unnecessarily stateful.